core/dbt/cli/requires.py

```diff
@@ -85,3 +87,67 @@ def wrapper(*args, **kwargs):
         return func(*args, **kwargs)
 
     return update_wrapper(wrapper, func)
+
+
+def runtime_config(func):
+    """A decorator used by click command functions for generating a runtime
+    config given a profile and project.
+    """
+
+    def wrapper(*args, **kwargs):
+        ctx = args[0]
+        assert isinstance(ctx, Context)
+
+        req_strs = ["profile", "project"]
+        reqs = [ctx.obj.get(req_str) for req_str in req_strs]
+
+        if None in reqs:
+            raise DbtProjectError("profile and project required for runtime_config")
+
+        ctx.obj["runtime_config"] = RuntimeConfig.from_parts(
+            ctx.obj["project"],
+            ctx.obj["profile"],
+            ctx.obj["flags"],
+        )
+
+        return func(*args, **kwargs)
+
+    return update_wrapper(wrapper, func)
+
+
+def manifest(*args0, write_perf_info=False):
```

Comment:
I think we would want to do something similar to what we did at line 55 of this file to allow people pass in a pre constructed manifest and skip the loading part, but that could be a separate issue we tackle later on.

CC @MichelleArk  thoughts?

Reply:
Yes, that makes sense. I thought this was being handled by the `dbtRunner`, but if that is not implemented in this decorator as well it will be overwritten every time.

Reply:
The dbtRunner only set it. The logic in these decorators are the ones that handles skip doing loading when things are set.